exchanges/sessions.py

```diff
@@ -0,0 +1,372 @@
+        datetime.strptime(data["data_provided_through_date"], "%Y-%m-%d")
+    )
+    # check date_ lies in overall date range
+    if not data_provided_from_date <= date_ <= data_provided_through_date:
+        logging.info(f"date {date_} lies outside of data range")
+        return []
+
+    # check if date_ is in non trading days
+    non_trading_days = get_non_trading_days(data)
+    if date__ in non_trading_days:
+        logging.info(f"date {date_} falls on non trading days")
+        return []
+
+    # check for any pause and resume events
+    if date__ in get_pause_days(data):
+        logging.info(f"date {date_} has a pause event")
+        sessions.append(
+            {
+                "timezone": data["timezone"],
+                "date": date_.strftime("%Y-%m-%d"),
+                "session": "",
+                "gte": {
+                    "time": data["intra_day_interruptions"][date_.date()]["hours"][0][
+                        "gte"
+                    ],
+                    "status": "PAUSE",
+                },
+                "lt": {
+                    "time": data["intra_day_interruptions"][date_.date()]["hours"][0][
+                        "lt"
+                    ],
+                    "status": "REOPEN",
+                },
+            }
+        )
+
+    # check if date_ is in partial trading days
+    if date__ in get_partial_trading_days(data):
+        logging.info(f"date {date_} falls on a partial trading day")
+        for partial_trading_hours in [
+            "standard_partial_early_trading_hours",
+            "standard_partial_regular_trading_hours",
+            "standard_partial_late_trading_hours",
+        ]:
+            for session in data[partial_trading_hours]:
+                sessions.append(
+                    {
+                        "timezone": data["timezone"],
+                        "date": date_.strftime("%Y-%m-%d"),
+                        "session": session["session"],
+                        "gte": {"time": session["gte"], "status": "OPEN"},
+                        "lt": {"time": session["lt"], "status": "CLOSE"},
+                    }
+                )
+        return sessions
+    # check if date is in irregular partial trading days
+    if date__ in get_irregular_partial_trading_days(data):
+        logging.info(f"date {date_} falls on an irregular partial trading day")
+        for session in data["irregular_partial_trading_days"][date_.date()]["sessions"]:
+            sessions.append(
+                {
+                    "timezone": data["timezone"],
+                    "date": date_.strftime("%Y-%m-%d"),
+                    "session": session["session"],
+                    "gte": {"time": session["hours"][0]["gte"], "status": "OPEN"},
+                    "lt": {"time": session["hours"][0]["lt"], "status": "CLOSE"},
+                }
+            )
+
+        return sessions
+    # check if date_ is on a weekday
```

Comment:
This code looks correct, but the comment shouldn't assume the regular trading days will be a weekday, so the logging info (and associated comment) should say "... does not fall on a regular trading day" rather than "...on a weekday".

Reply:
have modified the logging info